### ISSUE TITLE
enable flaky local cluster tests

### ISSUE THREAD

**Issue by t-neIson** [solana-labs-community/solana#71]
#### Problem
Some of tests in `local_cluster` package fails only in ci. This PR is going to be used to figure out the reason and fix the issue.

#### Summary of Changes
1. Enable flaky tests
2. Only run `stable` job for quick iteration

Fixes #5350

**Comment by sakeridge** [comment]
local_cluster/src/cluster_tests.rs
```rust
        for validator in &cluster_nodes {
            if ignore_nodes.contains(&validator.id) {
                continue;
        let sigResult =
```
there's usually a fundamental issue in the protocol when the signature isn't being confirmed, which needs to be root caused. Indeed the purpose of this function is to send a transaction to each specific validator, and to check that the transaction then propagates to the rest of the network. Thus, a failure to get the transaction signature on any iteration of this loop should fail the entire test, and we shouldn't retry with another client.

**Comment by t-neIson** [comment]
understood.

anyways, was just expertimenting with different things to get any useful debug information.

**Comment by sakeridge** [comment]
no problem!

**Comment by jeffweshington** [comment]
yep

**Comment by t-neIson** [comment]
rayon-threadlimit/src/thread_count.rs
```rust
    *MAX_RAYON_THREADS.read().unwrap()
}

pub fn init_test_thread_count() {
```
nit: how about:
```rust
pub fn set_thread_count(count: usize) ..
```
then do `set_thread_count(1)` from the one place that `init_test_thread_count()` is called instead

**Comment by sakeridge** [comment]
Yeah I was just closing it off. Didn't want it to be an accessible api. But a hack's a hack. I can change it to this.

**Comment by mvilnes** [comment]
I don't feel strongly :)

**Comment by CrieasofCarrots** [comment]
Any objections to landing this?

**Comment by jeffweshington** [comment]
r+ from me at least :)
